platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/EmbeddedSolrFiles.java

```diff
@@ -0,0 +1,178 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.solr.factory.impl;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Objects;
+import javax.xml.parsers.ParserConfigurationException;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang.Validate;
+import org.apache.solr.core.SolrConfig;
+import org.apache.solr.schema.IndexSchema;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+/**
+ * Maintain embedded Solr configuration and schema files.
+ *
+ * <p><i>Note:</i> The corresponding {@link SolrConfig} and {@link IndexSchema} will be constructed
+ * the first time they are retrieved.
+ */
+public class EmbeddedSolrFiles {
+  private static final Logger LOGGER = LoggerFactory.getLogger(EmbeddedSolrFiles.class);
+
+  private final String coreName;
+  private final String configName;
+  private final File configFile;
+  private final String schemaName;
+  private final File schemaFile;
+  private volatile SolrConfig solrConfig = null;
+  private volatile IndexSchema schemaIndex = null;
+
+  /**
+   * Constructor.
+   *
+   * @param coreName name of the Solr core
+   * @param configXml name of the Solr configuration file
+   * @param schemaXmls file names of the Solr core schemas to attempt to load (will start with the
+   *     first and fallback to the others in order if unavailable)
+   * @param configProxy {@link ConfigurationFileProxy} instance to use
+   * @throws IllegalArgumentException if <code>coreName</code>, <code>configXml</code>, <code>
+   *     schemaXmls</code>, or <code>configProxy</code> is <code>null</code> or if unable to find
+   *     any files
+   */
+  public EmbeddedSolrFiles(
```

Comment:
❓ Should this class and its methods be package-private instead of `public`?

Reply:
I won't be able to test them if they are package private 😄 
I can make the class package private though but nothing in this class is really harmful anyway. It is meant as a utility class to offload some of the complexity that was in the EmbeddedSolrFactory class before.

Reply:
Made the class package private